src/main/java/com/bean00/RequestParser.java

```diff
@@ -1,13 +1,64 @@
 package com.bean00;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class RequestParser {
+    private BufferedReader in;
+    private String requestLine;
+    private String requestURL;
+    private List<String> headers = new ArrayList<>();
+
+    public RequestParser(BufferedReader in) {
+        this.in = in;
+    }
+
+    public void parseRequest() throws IOException {
+        parseRequestLine();
+
+        parseHeaders();
+    }
+
+    public Request buildRequest() {
+        Request request = new Request(requestURL, headers);
+
+        return request;
+    }
+
+    public String getRequestURL() {
```

Comment:
This doesn't seem to belong. Is the `RequestParser` modeling the request? Isn't that the responsibility of the `Request`?

Reply:
I included those methods to be able to test `parseRequest` without using Mocks. I could rewrite the `parseRequest` tests to use Mocks instead.